### ISSUE TITLE
"Error: You must pass a string or Handlebars AST to Handlebars.compile- BUT HANDLEBARS WORKS (but is causing other error

### ISSUE THREAD

**Issue by arssonist** [wycats/handlebars.js#1291]
The Problem: Getting an error that says I need to pass a string to HB, and I'm instead passing undefined. This is happening on every page of the site, even ones with no templates. But the templates are WORKING on the two pages that I use them and I am passing strings.

The problem is that this error is causing some JQuery to fail on other pages where I am not using HB. I know this b/c If I comment out this code below, the JQ works properly on another page entirely unrelated to this code.
```
// // // 1. pull in template
var blogTemplateString = $('#blog-template').html();

// // // 2.Compile the string into a function
  var blogTemplateRenderer = Handlebars.compile(blogTemplateString);

// // // 3.Input response into the function
  var output = blogTemplateRenderer(data)
// //
// 4.append to page
  $('#blog-details').html(output);
```
It appears that HB is loading on everypage of the app, even ones where it isn't supposed to. I'm also using Sinatra and Foundation here. Are there known problems between these and HB?

UPDATE: Out of two, one template was ajax and the other was plain internal data. The non-ajax was the code above.
 I turned it into an Ajax and now the jquery is working again, however the HB errors are still persistent across the app. Something is weird here.

**Comment by nknapp** [comment]
If you are passing "undefined" to Handlebars, than the output is correct. I would place a `debugger` or a  `console.log()` statement in front of the `Handlebars.compile()`-call, in order to see what is actually passed to `.compile()`.

jQuery will return `undefined`, if there is a typo in the `#blog-template'-selector. Maybe that is the problem.

I don't believe that this is an error in Handlebars, so I'm closing this issue. You are still free to add comments. Would be nice to hear, if you solved your problem.

**Comment by nknapp** [comment]
Reopening, because it feels wrong to close this issue so quickly...

**Comment by arssonist** [comment]
So I do `console.log()` the output, and I've used breakpoints in Chrome debugger. And the output is correct! And this output is appending properly. I am happy with this.

But the problem is on the pages w/o any Handlebars. The HB is effecting these other pages and causing errors. 

Example: The Blog view has Handlebars. Index view does not, but is still getting a HB error in the console nonetheless, and this makes unrelated JQ not work. This JQ is unrelated to the HBs functionality, it's on a different view and the code is written on a different .js file. Yet the HB is still effecting it.

I know b/c if I comment out the HB code, and the unrelated index page begins to work again. I've found a hack where I get the reponseData with Ajax and this makes the unrelated page index display properly again.
But the errors are all still there! 

I think this might be a Sinatra problem, as it seems to be pulling in the `Ids` and `classes` from other templates in a kind of ghost way. U can't access them, but they are there in the background. This might be effecting the behaviour of HB and JQ. I'm not totally sure though.

For now the Ajax hack fixed the problem, even though  I don't need Ajax for this situation, and I feel like anything could break my functionality.  

It's bandaged but not fixed. Unless there is a freak chance someone is a Sinatra/Handlebars expert, and can help me truly solve the error, I would think U could close this.
